// @celo/contractkit/lib/base.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var CeloContract;
(function (CeloContract) {
    CeloContract["Accounts"] = "Accounts";
    CeloContract["Attestations"] = "Attestations";
    CeloContract["BlockchainParameters"] = "BlockchainParameters";
    CeloContract["DoubleSigningSlasher"] = "DoubleSigningSlasher";
    CeloContract["DowntimeSlasher"] = "DowntimeSlasher";
    CeloContract["Election"] = "Election";
    CeloContract["EpochRewards"] = "EpochRewards";
    CeloContract["Escrow"] = "Escrow";
    CeloContract["Exchange"] = "Exchange";
    CeloContract["FeeCurrencyWhitelist"] = "FeeCurrencyWhitelist";
    CeloContract["Freezer"] = "Freezer";
    CeloContract["GasPriceMinimum"] = "GasPriceMinimum";
    CeloContract["GoldToken"] = "GoldToken";
    CeloContract["Governance"] = "Governance";
    CeloContract["LockedGold"] = "LockedGold";
    CeloContract["MetaTransactionWallet"] = "MetaTransactionWallet";
    CeloContract["MetaTransactionWalletDeployer"] = "MetaTransactionWalletDeployer";
    CeloContract["MultiSig"] = "MultiSig";
    CeloContract["Random"] = "Random";
    CeloContract["Registry"] = "Registry";
    CeloContract["Reserve"] = "Reserve";
    CeloContract["SortedOracles"] = "SortedOracles";
    CeloContract["StableToken"] = "StableToken";
    CeloContract["TransferWhitelist"] = "TransferWhitelist";
    CeloContract["Validators"] = "Validators";
})(CeloContract = exports.CeloContract || (exports.CeloContract = {}));
exports.ProxyContracts = Object.keys(CeloContract).map(function (c) { return c + "Proxy"; });
exports.AllContracts = Object.keys(CeloContract);
var AuxiliaryContracts = [
    CeloContract.MultiSig,
    CeloContract.MetaTransactionWalletDeployer,
    CeloContract.MetaTransactionWallet,
];
exports.RegisteredContracts = exports.AllContracts.filter(function (v) { return !AuxiliaryContracts.includes(v); });
